plugin.video.embycon/resources/lib/cache_images.py

```diff
@@ -99,43 +99,82 @@ def cache_artwork_interactive(self):
             xbmc.executebuiltin('ActivateWindow(servicesettings)')
             return
 
-        # ask to delete all textures
-        question_result = xbmcgui.Dialog().yesno(string_load(30296), string_load(30297))
-        if question_result:
-            pdialog = xbmcgui.DialogProgress()
-            pdialog.create(string_load(30298), "")
+        result_report = []
+
+        # ask questions
+        question_delete_unused = xbmcgui.Dialog().yesno(string_load(30296), string_load(30297))
+        question_cache_images = xbmcgui.Dialog().yesno(string_load(30299), string_load(30300))
+
+        delete_canceled = False
+        # now do work - delete unused
+        if question_delete_unused:
+            delete_pdialog = xbmcgui.DialogProgress()
+            delete_pdialog.create(string_load(30298), "")
             index = 0
 
-            json_result = json_rpc('Textures.GetTextures').execute()
+            params = {"properties": ["url"]}
+            json_result = json_rpc('Textures.GetTextures').execute(params)
             textures = json_result.get("result", {}).get("textures", [])
-            log.debug("texture ids: {0}", textures)
-            total = len(textures)
-            for texture in textures:
-                texture_id = texture["textureid"]
-                params = {"textureid": int(texture_id)}
-                json_rpc('Textures.RemoveTexture').execute(params)
-                percentage = int((float(index) / float(total)) * 100)
-                message = "%s of %s" % (index, total)
-                pdialog.update(percentage, "%s" % (message))
 
-                index += 1
-                if pdialog.iscanceled():
-                    break
+            emby_texture_urls = self.get_emby_artwork(delete_pdialog)
 
-            del textures
-            del pdialog
+            log.debug("kodi textures: {0}", textures)
+            log.debug("emby texture urls: {0}", emby_texture_urls)
 
-        question_result = xbmcgui.Dialog().yesno(string_load(30299), string_load(30300))
-        if not question_result:
-            return
+            if emby_texture_urls is not None:
+
+                unused_texture_ids = set()
+                for texture in textures:
+                    url = texture.get("url")
+                    url = urllib.unquote(url)
+                    url = url.replace("image://", "")
+                    url = url[0:-1]
+                    log.debug("checking texture url: {0}", url)
```

Comment:
This is logged multiple times, reduce the log spamming.

Reply:
removed extra logging